### PR TITLE
KAA-1113 Updated "Documentation contribution and style guide"

### PR DESCRIPTION
**Pull request by balldir**

### CODE REVIEW

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

**Comment by forGGe** [comment]
+

File: doc/Customization-guide/Documentation-contribution-and-style-guide/index.md [hunk 1]
```diff
@@ -127,7 +119,7 @@ permalink: /:path/
 ```
 {% endraw %}
 
-* Refer to another page through root_url , e.g.,  [Glossary]({{{root_url}}Glossary).
+* Refer to another page through root_url , e.g.,  [Glossary]({{root_url}}Glossary).
```
Comment (rasendubi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md]: Put `root_url` in backticks. There is also extra space after it.

File: doc/Customization-guide/Documentation-contribution-and-style-guide/index.md [hunk 2]
```diff
@@ -37,11 +28,11 @@ This guide introduces documentation conventions applicable to the Kaa project. T
 ## Headings and capitalization
 * Capitalize only the first word in titles and headings.
 * Do not use capitalization for no apparent reason. Use lowercase unless uppercase is justified.
-* Use Heading 4 (e.g. ####) as the last level heading.
+* Use Heading 4 (e.g. ####) as the last level heading. All other levels are not displayed in TOC.
```
Comment (rasendubi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md]:
```
(e.g. ####)
```

->

```
(e.g., `####`)
```

Comment (rasendubi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md]: Should be `i.e.`, not `e.g.`

File: doc/Customization-guide/Documentation-contribution-and-style-guide/index.md [hunk 3]
```diff
@@ -186,19 +178,19 @@ Eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dic
 <div class="tab-content">
 <div id="Ubuntu" class="tab-pane fade in active" markdown="1" >
 
-### Ubuntu
+##### Ubuntu
```
Comment (rasendubi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md:191]: Different number of `#` in code and rendered (6 and 5).

File: doc/Customization-guide/Documentation-contribution-and-style-guide/index.md [hunk 4]
```diff
@@ -37,11 +28,11 @@ This guide introduces documentation conventions applicable to the Kaa project. T
 ## Headings and capitalization
 * Capitalize only the first word in titles and headings.
 * Do not use capitalization for no apparent reason. Use lowercase unless uppercase is justified.
-* Use Heading 4 (e.g. ####) as the last level heading.
+* Use Heading 4 (i.e. `####`) as the last level heading. All other levels are not displayed in table of contents.
 
 ## Technical terms and abbreviations
 * Spell out a technical abbreviation on its first mention on the page.
```
Comment (akokhanovskyi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md]: Please provide an example, e.g.:
_CTL_ (Common Type Library)

File: doc/Customization-guide/Documentation-contribution-and-style-guide/index.md [hunk 5]
```diff
@@ -67,13 +58,14 @@ This guide introduces documentation conventions applicable to the Kaa project. T
 ## Markdown formatting
 * Use [GitHub Flavored Markdown](https://guides.github.com/features/mastering-markdown/) for all documentation.
 * Do not use Emoji.
-* Use next template for TOC.
+* Place next template to auto generate table of contents.
```
Comment (akokhanovskyi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md]: Use the following template to auto-generate the table of page contents:

File: doc/Customization-guide/Documentation-contribution-and-style-guide/index.md (hunk 4 above)
Comment (akokhanovskyi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md]: Replace "technical abbreviation" with "acronym".

File: doc/Customization-guide/Documentation-contribution-and-style-guide/index.md [hunk 6]
```diff
@@ -67,13 +58,14 @@ This guide introduces documentation conventions applicable to the Kaa project. T
 ## Markdown formatting
 * Use [GitHub Flavored Markdown](https://guides.github.com/features/mastering-markdown/) for all documentation.
 * Do not use Emoji.
```
Comment (akokhanovskyi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md:70]: - Start every sentence with a new line to make future merging easier.

File: doc/Customization-guide/Documentation-contribution-and-style-guide/index.md (hunk 5 above)
Comment (rasendubi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md]: page table?

Comment (akokhanovskyi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md]: TOC for the page. Page TOC.

**Comment by balldir** [comment]
@akokhanovskyi @rasendubi @forGGe  re-review please

File: doc/Customization-guide/Documentation-contribution-and-style-guide/index.md [hunk 7]
```diff
@@ -211,19 +213,29 @@ Eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dic
 
 # Documentation contribution
 
-Kaa documentation is a part of [Kaa source code](https://github.com/kaaproject/kaa) - it can be found in the `doc/` folder. A full description of contributing to Kaa project can be found at [How to contribute]({{root_url}}Customization-guide/How-to-contribute/). Key differences between contributing source code and documentation are :
+Kaa documentation is a part of [Kaa source code](https://github.com/kaaproject/kaa) - it can be found in the `doc/` folder.
```
Comment (akokhanovskyi) [inline doc/Customization-guide/Documentation-contribution-and-style-guide/index.md:216]: Maybe make the "doc folder" a link to
https://github.com/kaaproject/kaa/tree/master/doc
? (It's not in master yet, but will be once we release).

**Comment by akokhanovskyi** [comment]
I think this is good for an iteration - let's merge so that everyone can easily see this.

**Pull request merged**